pkg/controllers/provisioningv2/fleetcluster/fleetcluster.go

```diff
@@ -43,10 +46,21 @@ func Register(ctx context.Context, clients *wrangler.Context) {
 		apply:         clients.Apply.WithCacheTypes(clients.Provisioning.Cluster()),
 	}
 
+	h.getPrivateRepoURL = func(cluster *v1.Cluster, mgmtCluster *mgmt.Cluster) string {
```

Comment:
just out of curiosity, why are you defining this function dynamically? I may be missing the context behind this.

Reply:
nevermind I see the test

Reply:
Yes, that's the reason I was hesitant to write a unit test in the first place.